.travis.yml

```diff
@@ -146,21 +146,21 @@ script:
 # libtorrent is the name of the test suite target
   - cd test
   - 'if [ "$tests" == "1" ]; then
-      travis_retry bjam -j3 warnings-as-errors=on ssl=$ssl crypto=$crypto debug-iterators=on picker-debugging=on invariant-checks=full $toolset variant=$variant libtorrent test_natpmp enum_if -l300 &&
+      travis_retry bjam -j3 c++-template-depth=512 warnings-as-errors=on ssl=$ssl crypto=$crypto debug-iterators=on picker-debugging=on invariant-checks=full $toolset variant=$variant libtorrent test_natpmp enum_if -l300 &&
```

Comment:
That's a good point actually. I didn't think to add it to the Jamfile because I don't experience this problem locally, but I believe that's because I use gcc 6. There's surely still lots of people using gcc 5 so this setting needs to be changed in the default build.

The offending statement is on line 130 in dht_tracker.cpp. The second `forward_as_tuple` call apparently triggers a rather deep nesting of template instantiations.